crates/project_panel/src/project_panel.rs

```diff
@@ -1226,6 +1335,24 @@ impl ProjectPanel {
         }
     }
 
+    fn should_omit_entry(snapshot: &Snapshot, entry: &Entry) -> bool {
+        if let Some(root_path) = snapshot.root_entry() {
+            if entry.path == root_path.path {
+                return false;
+            }
+        }
+
+        let child_entries: Vec<&Entry> = snapshot.child_entries(&entry.path).collect();
```

Comment:
I don't think we need to collect this entire vector each time, can we just check if 0, 1, or 2 entries exist and do this without the collect?